packages/gradle/src/plugin/dependencies.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  CreateDependenciesContext,
+  DependencyType,
+  RawProjectGraphDependency,
+  validateDependency,
+} from '@nx/devkit';
+import { requireNx } from '@nx/devkit/nx';
+import { readFileSync } from 'node:fs';
+import { basename } from 'node:path';
+
+import { getGradleReport } from '../utils/get-gradle-report';
+
+const {
+  createProjectRootMappingsFromProjectConfigurations,
+  findProjectForPath,
+} = requireNx();
+
+export const createDependencies = async (
+  _,
+  context: CreateDependenciesContext
+) => {
+  const { filesToProcess, projects } = context;
+
+  const gradleFiles = findGradleFiles(filesToProcess);
+
+  if (gradleFiles.length === 0) {
+    return [];
+  }
+
+  const projectRootMappings =
+    createProjectRootMappingsFromProjectConfigurations(projects);
+  let dependencies: RawProjectGraphDependency[] = [];
+  console.time('locating gradle dependencies');
+  const { gradleFileToGradleProjectMap, buildFileToDepsMap } =
+    getGradleReport();
+  /**
+   * Map of gradle project name to nx project name
+   */
+  const gradleProjectToNxProjectMap = new Map<string, string>();
+  for (const [buildFile, gradleProject] of gradleFileToGradleProjectMap) {
+    const nxProject = findProjectForPath(buildFile, projectRootMappings);
+    gradleProjectToNxProjectMap.set(gradleProject, nxProject as string);
+  }
+
+  for (const [source, gradleFile] of gradleFiles) {
+    const depsFile = buildFileToDepsMap.get(gradleFile) as string;
+
+    if (depsFile) {
+      dependencies = dependencies.concat(
+        processGradleDependencies(
+          depsFile,
+          gradleProjectToNxProjectMap,
+          source,
+          gradleFile,
+          context
+        )
+      );
+    }
+  }
+  console.timeEnd('locating gradle dependencies');
```

Comment:
Can we change this to performance.mark / performance.measure so they show up with NX_PERF_LOGGING?

Reply:
changed.